packages/react-server/src/ReactFlightServer.js

```diff
@@ -4089,6 +4104,8 @@ export function halt(request: Request, reason: mixed): void {
     if (request.destination !== null) {
       flushCompletedChunks(request, request.destination);
     }
+    const onAllReady = request.onAllReady;
+    onAllReady();
```

Comment:
If we abort we clear out all tasks so we should be for the first time hitting zero and can call onAllReady. If we don't do this here then something needs to ping to scheduleWork where it is tried but this can be very late or never come at all. I added here and to abort in case we end up shipping prerender without halt